Podcast/DiscoverTopicViewController.swift

```diff
@@ -170,10 +142,39 @@ class DiscoverTopicViewController: DiscoverComponentViewController {
             topicLabel.attributedText = topicString
         } else if let topicType = topic.topicType {
             // parent topic
-            topicImageView.image = topicType.headerImage
+            image = topicType.headerImage
             topicLabel.text = topic.name
         }
 
+        topicLabel.sizeToFit()
+        navigationItem.titleView = topicLabel
+        navigationController?.navigationBar.setBackgroundImage(image.resizableImage(withCapInsets: UIEdgeInsets.zero, resizingMode: .stretch), for: .default)
+
+        navigationController?.navigationBar.tintColor = .offWhite
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationItem.titleView = nil
+        navigationController?.navigationBar.setBackgroundImage(nil, for: .default)
+    }
```

Comment:
So this sometimes lags and doesn't change the background image right away. Use `override func willMove(toParentViewController parent: UIViewController?) ` to immediately clear it when the back button is pressed.

Reply:
Hmm well actually if you use that then when you press the back button in a subtopic view controller it will clear the nav bar for a sec and then re-insert the image... so maybe check if the parent is DiscoverViewController

Reply:
this doesn't work so I am going to keep it this way for now